src/modules/Cart/Cart.js

```diff
@@ -1,33 +1,64 @@
+import _ from 'lodash';
 import React from 'react';
 import cn from 'classnames';
 import { Link } from 'react-router-dom';
 import './Cart.scss';
+import CartItem from '../../components/CartItem/CartItem';
+import isEmpty from './../../validation/is-empty';
 
 class Cart extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      showCart: false
+      showCart: false,
     }
   }
+
+  renderCart = () => {
+    if (!isEmpty(this.props.products)) {
+      return _.map(this.props.cart, (quantity, key) => {
+        let productInfo = this.props.products[key];
+        let { name, price_n, images } = productInfo;
+        let product = { id: key, quantity, name, price_n, image: images[0] };
+        return <CartItem
+          key={key}
+          product={product}
+          removeProduct={this.props.removeProduct}
+        />
+      })
+    }
+    return false;
+  }
+
+  calculatePrice = () => {
+    if (!isEmpty(this.props.products)) {
+      let total = 0;
+      _.map(this.props.cart, (quantity, key) => {
```

Comment:
vì sao không dùng  this.props.cart.map

Reply:
Vì this.props.cart là object nên dùng lodash cho tiện a, khỏi phải viết Object.keys ...